### ISSUE TITLE
Regresion Logistica

### ISSUE THREAD

**Issue by vargasmiranda-victormanuel** [vargasmiranda-victormanuel/course-sistemas-basados-en-conocimiento#2]
Quiero crear un jupyter notebook que contenga una explicación del algoritmo de Regresion Logistica y solucion de un caso real.
Usar la libreria scikit-learn encontrada en https://scikit-learn.org/stable/install para resolver un caso real.
Debe incluir explicaciones claras y argumentadas sobre:
Qué problema resuelve
Supuestos fundamentales del modelo
Usos y aplicaciones potenciales
En qué tipos de problemas es apropiado
Cuándo es preferible frente a modelos más complejos
Pros y contras desde la perspectiva de XAI
Debe incluirse un análisis explícito de la explicabilidad del modelo.
Sobre la solucion tecnica:
Debe cargar el dataset público digits dataset y debe estar referenciado en el notebook. Este dataset se optiene de la pagina https://scikit-learn.org/stable/auto_examples/classification/plot_digits_classification.html y puede extraido con el siguiente codigo. de ejemplo:
from sklearn.datasets import load_digits
digits = load_digits()
X = digits.data
y = digits.target
Debe describirse brevemente la naturaleza del problema (qué se predice y por qué).
Limpieza de datos basica, solo si es necesaria
Selección de variables
Separación en conjuntos de entrenamiento y prueba
Implementación del algoritmo utilizando Scikit-Learn
Ajuste del modelo a los datos
Evaluación del modelo mediante uso de métricas apropiadas (por ejemplo: MSE, RMSE, R²)
Interpretación de los resultados obtenidos
Demostrar el uso del modelo entrenado sobre datos de prueba o nuevos datos

Descripcion del dataset

The digits dataset consists of 8x8 pixel images of digits. The images attribute of the dataset stores 8x8 arrays of grayscale values for each image. We will use these arrays to visualize the first 4 images. The target attribute of the dataset stores the digit each image represents and this is included in the title of the 4 plots below.
Note: if we were working from image files (e.g., ‘png’ files), we would load them using [matplotlib.pyplot.imread](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.imread.html#matplotlib.pyplot.imread).
digits =[ datasets.load_digits](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_digits.html#sklearn.datasets.load_digits)()

_, axes =[ plt.subplots](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.subplots.html#matplotlib.pyplot.subplots)(nrows=1, ncols=4, figsize=(10, 3))
for ax, image, label in zip(axes, digits.images, digits.target):
    ax.set_axis_off()
    ax.imshow(image, cmap=plt.cm.gray_r, interpolation="nearest")
    ax.set_title("Training: %i" % label)

# flatten the images
n_samples = len(digits.images)
data = digits.images.reshape((n_samples, -1))

# Create a classifier: a support vector classifier
clf =[ svm.SVC](https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html#sklearn.svm.SVC)(gamma=0.001)

# Split data into 50% train and 50% test subsets
X_train, X_test, y_train, y_test =[ train_test_split](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html#sklearn.model_selection.train_test_split)(
    data, digits.target, test_size=0.5, shuffle=False
)

# Learn the digits on the train subset
clf.fit(X_train, y_train)

# Predict the value of the digit on the test subset
predicted = clf.predict(X_test)

Below we visualize the first 4 test samples and show their predicted digit value in the title.
_, axes =[ plt.subplots](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.subplots.html#matplotlib.pyplot.subplots)(nrows=1, ncols=4, figsize=(10, 3))
for ax, image, prediction in zip(axes, X_test, predicted):
    ax.set_axis_off()
    image = image.reshape(8, 8)
    ax.imshow(image, cmap=plt.cm.gray_r, interpolation="nearest")
    ax.set_title(f"Prediction: {prediction}")
